benchmark.py

```diff
@@ -65,11 +64,6 @@ def compare(self, predicted, matchingFunc, output_fn):
                 found = False
                 
                 for predictedEx in predictedExtractions:
-                    if output_fn in predictedEx.matched:
-                        # This predicted extraction was already matched against a gold extraction
-                        # Don't allow to match it again
-                        continue
```

Comment:
This again, seems to have been over compensating for the matching bug. If a package returned the entire sentence in a predicted extraction, it could be used over and over again with the matching function returning 100% word overlap and the package would score 100% recall. Now that the matching function compares the predicate words and at least makes sure the predicate words overlap, it's unlikely you will encounter a situation where an extraction is used over and over again in that manner. 

Although generally the gold solutions **don't** split conjunctions ("I ate apples, orange, and pears" -> (I; ate, apples, orange, and pears)). There are _some_ cases where the gold solutions do ((I; ate, apples), (I; ate, oranges), (I; ate, pears)). By allowing these predicted extractions to now be re-used, we support the case where  (I; ate, apples, orange, and pears) can match multiple times, with (I; ate; apples) and then match with (I; ate, oranges), and then match with (I; ate; pears).

Reply:
I worry that this might allow systems spamming by repeats the extraction to get credit. 
I'm not sure I understand the explanation on why this is not the case?

Reply:
**An example why this can be removed without repeats being an issue:**

_Sentence:_
John went to the park while Sally drove to the store.

_Gold Extractions would be:_
(John; went to; the park)
(Sally; drove to; the park)

Before this change the package could could have a single bad extraction output like:
(John; went to the park while Sally drove to the store) - 100% word overlap

and it would only get counted once for (John; went to; the park), but not for (Sally; drove to; the park) due to the the code that checks if an extraction is already matched.

With the matching bug fixed, since the predicate must also match, even when you remove this checking code if an extraction is already matched it _still_ only gets counted once since the predicate only matches (John; went to; the park) and not (Sally; drove to; the park).

Therefore, a package with a singular long, lengthy bad extraction with a lot of word overlap with many gold extractions with this matching bug fixed doesn't get to claim credit over and over again for all of the gold extractions, but only with those that have the same predicate. This is why it should be safe to remove this check.

**A reason why this should be removed:**
See lines 8212-8215 in `all.oie`:
```
In Jewish Hebrew , the Samaritans are called `` Shomronim '' , while in Samaritan Hebrew they call themselves `` Shamerim '' .	called	called	the Samaritans	Shomronim
In Jewish Hebrew , the Samaritans are called `` Shomronim '' , while in Samaritan Hebrew they call themselves `` Shamerim '' .	called	called	the Samaritans	Shamerim
In Jewish Hebrew , the Samaritans are called `` Shomronim '' , while in Samaritan Hebrew they call themselves `` Shamerim '' .	call	called	the Samaritans	Shomronim
In Jewish Hebrew , the Samaritans are called `` Shomronim '' , while in Samaritan Hebrew they call themselves `` Shamerim '' .	call	called	the Samaritans	Shamerim
```

The 4 gold extractions are:
(the Samaritans; called; Shomronim)
(the Samaritans; called; Shamerim)
(the Samaritans; called; Shomronim)
(the Samaritans; called; Shamerim)

There are 2 duplicates in the gold solution. If you don't allow the system's extractions to count multiple times in this case for example, they will be penalized.